src/App/screens/Instructor/state/reducers/index.js

```diff
@@ -1,7 +1,7 @@
 import {combineReducers} from 'redux';
 import * as instructorActionTypes from '../actions/instructorActionTypes';
 
-const instructor = (state = false, action) => {
+const instructor = (state = {firstLessonState: 'approved'}, action) => {
```

Comment:
I don't know that you'd need to setState to compute it locally. This page is pretty much static. Pass in lessons as props and render. 

Even if you were polling the server for updates, it would mean new lessons loaded in and presented in stateless manner. Am I missing something?

A single source of truth is meant to be a single source of truth. It's not a primary source of some of the truth. I'd definitely vote for sticking to that without diversion for convenience, and see how it pans out.